### ISSUE TITLE
Getting error when using !GetAtt for SecurityGroups Property of type AWS::AutoScaling::LaunchConfiguration

### ISSUE THREAD

**Issue by nwesoccer** [aws-cloudformation/cfn-python-lint#1500]
*cfn-lint version: 0.29.6*

*Description of issue.*
We are getting the following error:

> cfnlint: E3008  Property "SecurityGroups" has no valid Fn::GetAtt options at Resources/TempLaunchConfig/Properties/SecurityGroups/0/Fn::GetAtt

The correlating piece of the template is:

```
TestLaunchConfig:
    Type: "AWS::AutoScaling::LaunchConfiguration"
    Properties:
      SecurityGroups:
        - !GetAtt TempSecurityGroup.GroupId
```

We do realize that we can use `!Ref TempSecurityGroup`, but because we have this in a lot of templates, we'd rather not change them all and continue using `!GetAtt TempSecurityGroup.GroupId` as it's supported by CloudFormation.

Relevant documentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-launchconfig.html#cfn-as-launchconfig-securitygroups

Related Issue:
https://github.com/aws-cloudformation/cfn-python-lint/issues/888

Cfn-lint should support passing in a GroupId using !GetAtt for the SecurityGroups Property list.  This template is valid and does deploy.

**Comment by PatMyron** [comment]
https://github.com/aws-cloudformation/cfn-python-lint/blob/b788cc9bd3d49ed20d5f2e58602755a0ef37f52c/src/cfnlint/rules/resources/properties/ValueRefGetAtt.py#L147

@nwesoccer what resource type is `TempSecurityGroup`? Is it in the same template?

**Comment by nwesoccer** [comment]
@PatMyron The type is `AWS::EC2::SecurityGroup`.  It is in the same template yes.

**Comment by PatMyron** [comment]
got it, able to reproduce, [`AWS::EC2::SecurityGroup.GroupId`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html#aws-properties-ec2-security-group-return-values) looks valid in the [official resource specification](https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json):
```json
    "AWS::EC2::SecurityGroup": {
      "Attributes": {
        "GroupId": {
          "PrimitiveType": "String"
        },
```
```yaml
Resources:
  TempSecurityGroup:
    Type: AWS::EC2::SecurityGroup
    Properties: 
      GroupDescription: GroupDescription
  TestLaunchConfig:
    Type: "AWS::AutoScaling::LaunchConfiguration"
    Properties:
      SecurityGroups:
        - !GetAtt TempSecurityGroup.GroupId
```

probably something to do with these `ExtendedSpecs` beyond the [official resource specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html):

https://github.com/aws-cloudformation/cfn-python-lint/blob/67fc5bb210b020e3226261f966a01726d574475d/src/cfnlint/data/ExtendedSpecs/all/04_property_values/aws_autoscaling.json#L65-L72

https://github.com/aws-cloudformation/cfn-python-lint/blob/7a21425545075a349b14c593104ee3aa13b8911a/src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_ec2.json#L80-L108

**Comment by kddejong** [comment]
We do have an option for name and/or GroupId then we can change it as needed.

https://github.com/aws-cloudformation/cfn-python-lint/blob/7a21425545075a349b14c593104ee3aa13b8911a/src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_ec2.json#L47-L79

**Comment by nwesoccer** [comment]
@kddejong It's the opposite the `!GetAtt TempSecurityGroup.GroupId` doesn't work, but name does.

**Comment by nwesoccer** [comment]
@kbcaleb @PatMyron Thank you both for the speedy communication and fix!

**Comment by chuxau** [comment]
Looks like there has been a regression back to this same issue 

* getting the following for every usage of !GetAtt for a Security Group defined in the same template
* `cfn-lint 0.83.7` on ubtuntu 20.04 installed via pip

```
[cfn-lint] E3008: Property "SourceSecurityGroupId" can Fn::GetAtt to a resource of types [AWS::EC2::SecurityGroup, AWS::EC2::VPC, AWS::EKS::Cluster] at Resources/dbIcmpIngressSGRule/Properties/SourceSecurityGroupId/Fn::GetAtt
[cfn-lint] E1010: Invalid GetAtt appEBAccessSG.GroupId for resource dbPingIngressSGRule
```
